mantidproject/mantid
# WISH missing reflection due to tube gap

**This must not be merged before #19207**

WISH users reported missing a reflection when using `PredictPeaks`. Current solution for now is to use a `tube-gap` parameter like CORELLI. 

**To test:**
 - The following script should fail before these changes and pass afterwards.

```python
import numpy as np

ws = LoadRaw(Filename='/archive/ndxwish/Instrument/data/cycle_16_5/WISH00038237.raw', OutputWorkspace='38237')
ws = ConvertUnits(ws, 'dSpacing', OutputWorkspace='38237')
UB = np.array([[-0.00601763,  0.07397297,  0.05865706],
                         [ 0.05373321,  0.050198,   -0.05651455],
                         [-0.07822144,  0.0295911,  -0.04489172]])
                         
SetUB(ws, UB=UB)

peaks = PredictPeaks(ws, WavelengthMin=0.1, WavelengthMax=100)
assert peaks.rowCount() == 510
filtered = FilterPeaks(peaks, "h^2+k^2+l^2", 75, '=')
assert filtered.rowCount() == 6
peak = filtered.row(2) 
assert  peak['h'] == -5 and peak['k'] == -1 and peak['l'] == -7
```

 - Check the new system test is passing (basically does the same as the above script)
 - Code review

Fixes #19315 .

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
